presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -787,6 +787,11 @@ protected RowExpression visitSubscriptExpression(SubscriptExpression node, Void
             RowExpression base = process(node.getBase(), context);
             RowExpression index = process(node.getIndex(), context);
 
+            // this block will handle row subscript, converts the ROW_CONSTRUCTOR with subscript to a DEREFERENCE expression
+            if (base instanceof SpecialFormExpression && base.getType() instanceof RowType) {
```

Comment:
You also need to check if `base` has form `ROW_CONSTRUCTOR`. I don't think `base.getType() instanceof RowType` is necessary though

Reply:
@highker  we need to check for RowType given that there can be two SpecialFromExpression can come here
1. ROW_CONSTRUCTOR with some index eg. row(1,2)[2]
2. DEREF(ROW_CONSTRUCTOR) with index eg. row(1, row(1,2))[2][2]
and also having an or wont work as DEREF of map can come here in which case the index is not constant ex

Reply:
Then keeping `base.getType() instanceof RowType` itself is good enough. The base is not necessary to be a `SpecialFormExpression`